msal/src/main/java/com/microsoft/identity/client/internal/controllers/MSALControllerFactory.java

```diff
@@ -22,25 +22,143 @@
-        //If Broker eligible return the BrokerMSALController
-        return new LocalMSALController();
+    /**
+     * Returns the appropriate MSAL Controller depending on Authority, App and Device state
+     *
+     * 1) The client indicates it wants to use broker
+     * 2) If not AAD Authority use local controller
+     * 3) If the the authority is AAD and the Audience is instance of AnyPersonalAccount
+     *    Use the local controller
+     * 4) If broker is not installed use local controller
+     * 5) Otherwise return broker controller
+     *
+     * @return
+     */
+    public static MSALController getAcquireTokenController(Context applicationContext, Authority authority, PublicClientApplicationConfiguration applicationConfiguration) {
+
+        if(brokerEligible(applicationContext, authority, applicationConfiguration)){
+            return new BrokerMSALController();
+        }else{
+            return new LocalMSALController();
+        }
+
     }
 
-    public List<MSALController> getAcquireTokenSilentControllers() {
-        //TODO: Add code to check if broker available and current app eligible to use the broker;
-        //If Broker eligible return both controller with local first then broker
+    /**
+     * Returns one or more controllers to address silent requests
+     *
+     * The order of the response matters.  The local controller should be returned first in order to
+     * ensure that any local refresh tokens are preferred over the use of the broker
+     *
+     * Only return the broker controller when the following are true:
+     *
+     * 1) The client indicates it wants to use broker
+     * 2) The authority is AAD
+     * 3) The audience is not AnyPersonalAccount
+     * 4) The broker is installed
+     * 5) The broker redirect URI for the client is registered
+     * @return
+     */
+    public static List<MSALController> getAcquireTokenSilentControllers(Context applicationContext, Authority authority, PublicClientApplicationConfiguration applicationConfiguration) {
+
         List<MSALController> controllers = new ArrayList<MSALController>();
         controllers.add(new LocalMSALController());
-        //controllers.add(new BrokerMSALController());
+        if(brokerEligible(applicationContext, authority, applicationConfiguration)) {
+            controllers.add(new BrokerMSALController());
+        }
 
         return controllers;
 
     }
+
+    /**
+     * Determine if request is eligible to use the broker
+     *
+     * Client indicates that it wants to use broker
+     * Authority == AzureActiveDirectoryAuthority
+     * Audience != AnyPersonalAccounts
+     * Broker Installed & Verified
+     *
+     *
+     * @param applicationContext
+     * @param authority
+     * @param applicationConfiguration
+     * @return
+     */
+    public static boolean brokerEligible(Context applicationContext, Authority authority, PublicClientApplicationConfiguration applicationConfiguration){
```

Comment:
Not an issue per se, but might consider naming this method `isBrokerEligible()`

Reply:
Ok.